### PR TITLE
[tests-only]tests: add tests for unlocking using file id

### PR DESCRIPTION
**Pull request by SwikritiT**
## Related Issue

- Part of: https://github.com/owncloud/ocis/issues/7625

### CODE REVIEW

**Review by ScharfViktor — APPROVED** [review]

File: tests/acceptance/features/bootstrap/WebDavLockingContext.php [hunk 1]
```diff
@@ -839,6 +885,34 @@ public function waitForCertainSecondsToExpireTheLock(int $time): void {
 		\sleep($time);
 	}
 
+	/**
+	 * @Then :count locks should be reported for file :file inside the space :space of user :user by the WebDAV API
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDavLockingContext.php:889]: should be adequate? 
```suggestion
	 * @Then :count locks should be reported for file :file inside the space :space of user :user
```

File: tests/acceptance/features/bootstrap/WebDavLockingContext.php [hunk 2]
```diff
@@ -174,6 +174,21 @@ public function userLocksFileInProjectSpaceUsingWebDavAPI(string $user, string $
...
+	 * @param string $user
+	 * @param string $file
+	 * @param string $space
+	 * @param TableNode $properties table with no heading with | property | value |
+	 *
+	 * @return void
+	 */
+	public function userHasLockedFileInProjectSpaceUsingWebDavAPI(string $user, string $file, string $space, TableNode $properties): void {
+		$this->userLocksFileInProjectSpaceUsingWebDavAPI($user, $file, $space, $properties);
+		$this->featureContext->theHTTPStatusCodeShouldBe(200, '');
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDavLockingContext.php:189]: I would suggest this flow
```suggestion
		$response = $this->userLocksFileInProjectSpace($user, $file, $space, $properties);
		$this->featureContext->theHTTPStatusCodeShouldBe(200, '', $response);
```

File: tests/acceptance/features/bootstrap/WebDavLockingContext.php [hunk 3]
```diff
@@ -839,6 +885,34 @@ public function waitForCertainSecondsToExpireTheLock(int $time): void {
...
+	 * @param string $spaceName
+	 * @param string $user
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function numberOfLockShouldBeReportedInProjectSpace(int $count, string $file, string $spaceName, string $user) {
+		$this->spacesContext->userSendsPropfindRequestToSpace($user, $spaceName, $file);
+		$this->featureContext->theHTTPStatusCodeShouldBe(207, "");
+		$responseXml = $this->featureContext->getResponseXml();
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDavLockingContext.php:902]: I would suggest this flow here as well
```suggestion
		$response = $this->spacesContext->userSendsPropfindRequestToSpace($user, $spaceName, $file);
		$this->featureContext->theHTTPStatusCodeShouldBe(207, "", $response);
		$responseXml = $this->featureContext->getResponseXml($response);
```

File: tests/acceptance/features/bootstrap/WebDavLockingContext.php (hunk 1 above)
Comment (SwikritiT) [reply tests/acceptance/features/bootstrap/WebDavLockingContext.php:889]: yes but we also already have step like `And 0 locks should be reported for file "textfile.txt" of user "Alice" by the WebDAV API` so I was just following the existing pattern. what do I do?

File: tests/acceptance/features/bootstrap/WebDavLockingContext.php (hunk 3 above)
Comment (SwikritiT) [reply tests/acceptance/features/bootstrap/WebDavLockingContext.php:902]: For this I'll make a follow up refactoring PR as the function `userSendsPropfindRequestToSpace` is used by many other functions so will be a lot of unrelated changes for this PR

File: tests/acceptance/features/bootstrap/WebDavLockingContext.php (hunk 1 above)
Comment (saw-jan) [reply tests/acceptance/features/bootstrap/WebDavLockingContext.php:904]: if it's new step then I would like to go with the suggestion

**Review by saw-jan — APPROVED** [review]

**Pull request merged**